initial pass at splitting the project into modules

agaro1121 opened the pull request.

markglh commented on build.sbt:
```diff
@@ -2,72 +2,63 @@
   project
-    .in(file("."))
+    .in(file("core"))
+    .settings(name := "core")
```
can we nest them all in `modules/xxx`

markglh replied:
I think it would be a little cleaner to have all the code modules nested

agaro1121 replied:
done

markglh commented on build.sbt:
```diff
@@ -2,72 +2,63 @@
-      "ch.qos.logback" % "logback-classic" % "1.2.3" % Compile
+val producer =
+  project
+    .in(file("producer"))
```
as above :)

agaro1121 replied:
done

markglh commented on consumer/src/main/scala/com/weightwatchers/reactive/kinesis/stream/Kinesis.scala:
```diff
@@ -0,0 +1,152 @@
+/**
+  * Main entry point for creating a Kinesis source and sink.
+  */
+object Kinesis extends LazyLogging {
```
Is it worth renaming to make it clear which is the consumer and which is the producer (vs just the package?) - I'm torn.

agaro1121 replied:
i think it's worth renaming. Don't ask me what that name should be though. Naming things is for the high wizards LOL

This is a tough one though.
It's a combination of consumer/producer + stream + source/sink
Looks like the old factory pattern to me. Should we go around those lines?

agaro1121 replied:
I named these:
ConsumerStreamFactory and ProducerStreamFactory
I'm open to suggestions

agaro1121 commented on producer/src/main/scala/com/weightwatchers/com/reactive/kinesis/stream/KinesisSinkGraphStage.scala:
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.weightwatchers.reactive.kinesis.stream
```
oops! Fixed!

agaro1121 commented:
@markglh I addressed your comments regarding the modules directory

# TODO
- The consumer/producer stream factories still need renaming
- `it` is its own module now. That solved a lot of compile issues. Thank you.

FYI Intellij does not like these changes LOL

agaro1121 commented on build.sbt:
```diff
@@ -2,72 +2,77 @@
+    .dependsOn(core)
+    .enablePlugins(AutomateHeaderPlugin, GitVersioning, ScalafmtCorePlugin)
+    .settings(settings)
+    .configs(IntegrationTest)
```
we'll probably need to get rid of this here and the others right?

agaro1121 commented on modules/it/src/it/scala/com/weightwatchers/reactive/kinesis/common/KinesisSuite.scala:
```diff
@@ -26,8 +23,10 @@ import scala.concurrent.duration._
+
   val defaultKinesisConfig: Config =
-    ConfigFactory.parseFile(new File("src/main/resources/reference.conf")).getConfig("kinesis")
+    ConfigFactory.parseFile(new File("modules/core/src/main/resources/reference.conf")).getConfig("kinesis")
```
@markglh having a weird issue here where this fails to load at the right time during testing
If i copy the contents of this ref.conf and paste them into the `it`'s local resource directory, it works fine.
Is this worth digging into?

agaro1121 commented on project/Dependencies.scala:
```diff
@@ -0,0 +1,77 @@
+  * These libs are included with the Amazon dependencies
+  * Our core module does not include amazon dependencies so we need to bring them in manually
+  * These versions match the ones in the Amazon dependencies
+  * */
```
@markglh Not sure how future-proof this is as we continue to update Amazon libs
Is there a better way to do this?

agaro1121 commented on build.sbt:
```diff
@@ -2,73 +2,71 @@
+  project
+    .in(file("."))
+    .settings(name := "reactive-kinesis")
+    .aggregate(core, consumer, producer)
```
I didn't include `it` here
i'm assuming we don't need to
